core/job/service/job_service.go

```diff
@@ -286,15 +306,35 @@ func (j *JobService) ReplaceAll(ctx context.Context, jobTenant tenant.Tenant, sp
 	updatedJobs, err := j.bulkUpdate(ctx, tenantWithDetails, toUpdate, logWriter)
 	me.Append(err)
 
-	err = j.bulkDelete(ctx, jobTenant, toDelete, logWriter)
+	deletedJobNames, err := j.bulkDelete(ctx, jobTenant, toDelete, logWriter)
 	me.Append(err)
 
 	err = j.resolveAndSaveUpstreams(ctx, jobTenant, logWriter, addedJobs, updatedJobs)
 	me.Append(err)
 
+	err = j.uploadJobs(ctx, jobTenant, addedJobs, updatedJobs, deletedJobNames)
+	me.Append(err)
+
 	return me.ToErr()
 }
 
+func (j *JobService) uploadJobs(ctx context.Context, jobTenant tenant.Tenant, addedJobs, updatedJobs []*job.Job, deletedJobNames []job.Name) error {
+	if len(addedJobs) == 0 && len(updatedJobs) == 0 && len(deletedJobNames) == 0 {
```

Comment:
We should just return if there are no jobs to upload. No need to raise error.

Reply:
updated